### ISSUE TITLE
Consider not building CMake on distros with a newer version available

### ISSUE THREAD

**Issue by SwooshyCueb** [irods/externals#265]
As we move towards `find_package` for more and more dependencies, the complexity that would be introduced by implementing distro-specific packaging in our current externals build system (and the amount of work required to implement it) is reduced. From where I stand today, I'm still uncertain whether it will *ever* be worth the effort to implement this for most packages, given that we will eventually be moving away from this system anyway.

However, I believe CMake is an exception. At present, switching to distro-provided CMake for distros with a newer version available than what we provide would involve making significantly fewer cross-repository changes than any other package. Moreover, we already use distro-provided CMake for newer distros in a number of places. Let's consider omitting CMake from externals in these cases.

**Comment by trel** [comment]
Yes - this makes a lot of sense.   We have not leaned on a new CMake feature/bugfix in quite some time.

**Comment by SwooshyCueb** [comment]
Superseded by irods/irods#8330
